Improve LogixNG action WindowToFront

danielb987 opened the pull request:
@dsand47 

This PR fixes: https://groups.io/g/jmriusers/message/228861

![Skärmbild_2024-04-05_00-41-28](https://github.com/JMRI/JMRI/assets/20255317/6c331a50-5f2e-444c-8b05-adf61de61695)

![Skärmbild_2024-04-05_00-40-57](https://github.com/JMRI/JMRI/assets/20255317/bd4e869f-2792-4daf-9fef-a86fca4a7fee)

danielb987 commented on java/src/jmri/jmrit/display/logixng/LogixNGBundle.properties:
```diff
@@ -101,3 +104,15 @@ LogixNG action:\n\
 Module: {1}\n\
 Action: {2}\n
 
+WindowToFront_HideOrShow_Show       = Show window
+WindowToFront_HideOrShow_Hide       = Hide window
+WindowToFront_HideOrShow_DoNothing  =
+
+WindowToFront_MaximizeMinimizeNormalize_Minimize    = Minimize window
+WindowToFront_MaximizeMinimizeNormalize_Normalize   = Normalize window
+WindowToFront_MaximizeMinimizeNormalize_Maximize    = Maximize window
+WindowToFront_MaximizeMinimizeNormalize_DoNothing   =
+
+WindowToFront_BringToFrontOrBack_Front      = Bring to front
+WindowToFront_BringToFrontOrBack_Back       = Bring to back
+WindowToFront_BringToFrontOrBack_DoNothing  =
```
@dsand47 
All these three selections have the option `Do nothing`. I wasn't sure if it's best to have the option empty or to have the text `Do nothing`. I'm open to both.

dsand47 replied:
The empty selection for do nothing works for me.

dsand47 commented:
The default actions need to be the blank entry, which should be the first entry.

The Action name, Window to front, is no longer accurate, although Show and Normalize also bring the window to the front.

danielb987 commented:
> The default actions need to be the blank entry, which should be the first entry.
> 
> The Action name, Window to front, is no longer accurate, although Show and Normalize also bring the window to the front.

Thanks. Both are fixed.

dsand47 commented:
I wonder if "Window Management" or "Window Actions" would be better than just plain Window.

If the window does not exist at run time a message is on the console:
```
22:09:54,731 jmri.jmrit.display.logixng.Window     ERROR - ddd [JMRI LogixNGThread]
```

And an error dialog:
![Screenshot 2024-04-05 at 10 10 44 PM](https://github.com/JMRI/JMRI/assets/17280992/1759975e-17c0-4f12-86ee-e33e07310cd0)

danielb987 commented:
> I wonder if "Window Management" or "Window Actions" would be better than just plain Window.
> 
> If the window does not exist at run time a message is on the console:

I have renamed the action to `Window management` and I have added the option Ignore `Window not found`.
The error message `ddd` was a left over from a debug session and I have removed it now.

![Skärmbild_2024-04-06_08-20-07](https://github.com/JMRI/JMRI/assets/20255317/a6ebe4fa-c871-40d6-834c-433b1ba73f0c)

dsand47 commented on help/en/releasenotes/current-draft-note.shtml:
```diff
@@ -436,9 +436,12 @@
+              See <a href="https://www.jmri.org/help/en/html/tools/logixng/reference/chapter13.shtml">
+              LogixNG Reference Chapter 13</a> for details.</li>
+          <li>The action <strong>Window to front</strong> has been improved and renamed to
+              <strong>Window</strong>. It can now show/hide a window, maximize/normalize/minimize
```
"Window" needs to be "Window management".

danielb987 replied:
Thanks. Fixed.

dsand47 reviewed (approved):
Release note change not required but recommended.
